ert/storage/_storage.py

```diff
@@ -46,13 +47,187 @@ class Config:
     record_type: ert.data.RecordType
 
 
-def _assert_server_info() -> None:
-    global _STORAGE_URL, _STORAGE_TOKEN  # pylint: disable=global-statement
+class StorageInfo:
+    _url: str = ""
+    _token: str = ""
 
-    if _STORAGE_URL is None:
+    @classmethod
+    def _set_info(cls) -> None:
         info = get_info()
-        _STORAGE_URL = info["baseurl"]
-        _STORAGE_TOKEN = info["auth"][1]
+        cls._url = info["baseurl"]
+        cls._token = info["auth"][1]
+
+    @classmethod
+    def reset(cls) -> None:
+        cls._url = ""
+        cls._token = ""
+
+    @classmethod
+    def url(cls) -> str:
+        if not StorageInfo._url:
+            cls._set_info()
+        return StorageInfo._url
+
+    @classmethod
+    def token(cls) -> str:
+        if not StorageInfo._token:
+            cls._set_info()
+        return StorageInfo._token
+
+
+class StorageRecordTransmitter(ert.data.RecordTransmitter):
+    _TYPE: ert.data.RecordTransmitterType = ert.data.RecordTransmitterType.ert_storage
+
+    def __init__(self, name: str, storage_path: str):
+        super().__init__()
+        self._name: str = name
+        self._storage_url = storage_path
+        self._uri: str = ""
+        self._record_type: Optional[ert.data.RecordType] = None
+
+    def _set_transmitted(self, uri: str, record_type: ert.data.RecordType) -> None:
+        super()._set_transmitted_state()
+        self._uri = uri
+        self._record_type = record_type
+
+    @property
+    def transmitter_type(self) -> ert.data.RecordTransmitterType:
+        return self._TYPE
+
+    async def _transmit(self, record: ert.data.Record) -> None:
+        record_type = record.record_type
+        if record_type != ert.data.RecordType.BYTES:
+            url = f"{self._storage_url}/{self._name}/matrix"
+        else:
+            url = f"{self._storage_url}/{self._name}/file"
+
+        await add_record(url, record)
+        self._set_transmitted(
+            f"{self._storage_url}/{self._name}", record_type=record_type
+        )
+
+    async def transmit_data(
+        self,
+        data: ert.data.record_data,
+    ) -> None:
+        if self.is_transmitted():
+            raise RuntimeError("Record already transmitted")
+        return await self._transmit(ert.data.make_record(data))
```

Comment:
I see the `make_record` call is here, and not so easy to replace, since we don't really know what the data type is.

Reply:
The problem lies in the design of the RecordTransmitter class which transmits not Records but data.....

Reply:
Again not really easy to avoid using it here, like it is used in all the other RecordTransmitter instances defined in `ert.data.record `

Reply:
I agree, maybe we need an issue for that.